test/abstracts.md

```diff
@@ -0,0 +1,107 @@
+# Abstract
+
+The following content represents ideal steps
+performed by hyperHTML to grant performance,
+recycle everything it can, and update layouts.
+
+It's rather a specification than current implementation,
+but the goal is to optimize such specification as much as possible
+and achieve best performance on top of that.
+
+```
+hyperHTML(statics, ...interpolations)
+│
+└▶  is `statics` known?
+    │
+    ├▶  YES
+    │
+    │   is current `context` representing `statics` ?
+    │   │
+    │   ├▶  YES
+    │   │
+    │   │   update through `interpolations`
+    │   │
+    │   └▶  NO  ┌─────────────────────────────────────────┐
+    │           ▼                                         │
+    │       clone the `fragment` related to `statics`     │
+    │                                                     │
+    │       retrieve the list of `paths` to update        │
+    │                                                     │
+    │       create a list of `updates`                    │
+    │                                                     │
+    │       relate `updates` to current `context`         │
+    │                                                     │
+    │       update `fragment` through `interpolations`    │
+    │                                                     │
+    │       replace `context` content with the `fragment` │
+    │                                                     │
+    └▶  NO                                                │
+                                                          │
+        create offline `fragment` with `statics`          │
+                                                          │
+        create a list of `paths` to update                │
+                                                          │
+        relate the `fragment` to `statics`                │
+                                                          │
+        relate the `paths` to `statics` ──────────────────┘
+```
+
+### Weakly referenced
+
+TODO: is an `expando` property that faster than a `WeakMap`?
+
+```
+context
+  ├▶  statics
+  └▶  updates
+
+any
+  └▶  wires
+```
+
+### Strongly referenced
+
+Template literals are _forever_, there's no reason to create extra GC pressure.
+
+TODO: is a native `Map` faster than a pair of arrays?
+
+```
+statics
+  ├▶  fragment
+  └▶  paths
+```
+
+### What is a wire ?
+
+A wire is a weakly linked document fragment content representing the node itself,
+as opposite of the container, like it is for `hyperHTML` bound `contexts`.
+
```

Comment:
Unless I've misunderstood, when you "wire" in hyperHTML, that isn't really used as a noun. It's a verb... in that, you wire `x` to `y`.  So, maybe:
 
> hyperHTML allows you to weakly associate a content object with a document fragment. Conceptually, this "wires" together a content object and a document fragment into a weak 1-to-1 relationship. The benefit of wiring are two fold: 
>
> 1. the weakly referenced content object can be retrieved from memory in constant time O(1), without the need to duplicate the object, 
> 2. because the reference to the content object is weak, the content object can be garbage collected by the JS Engine when it is no longer needed by the application.